headers.py

```diff
@@ -138,6 +140,27 @@ def checkoutArtifact(experimentName, trialNum, commitHash, fileName):
     os.chdir(original_dir)
     return res
 
+def fork(experimentName, commitHash, xp_state : State, outputDir = None):
+	original_dir = os.getcwd()
+	if outputDir is None:
+		outputDir = original_dir
+	outputDir = os.path.abspath(outputDir)
+	if outputDir == original_dir:
+		output = subprocess.check_output('git status'.split()).decode().splitlines()
+		if output[1] != 'nothing to commit, working directory clean':
+			raise Error("Cannot fork from a dirty working directory")
+			#are we assuming that the user's directory is using git at the time?
```

Comment:
We should do some try catch. If it's current dir is not a git repo, then we should also raise error. You don't want to wipe a system directory by accident.